src/jit/morph.cpp

```diff
@@ -2101,7 +2104,8 @@ GenTreePtr    Compiler::fgMakeTmpArgNode(unsigned tmpVarNum
 
         arg->gtFlags |= GTF_DONT_CSE;
         arg = gtNewOperNode(GT_ADDR, TYP_BYREF, arg);
-        
+        varDsc->lvAddrExposed = 1;
```

Comment:
Instead of doing this in four places we could instead check for a GT_ADDR at the end of thios method before the return arg;

if (arg->gtOper == GT_ADDR)
{
    lvaSetVarAddrExposed(tmpVarNum);        // This will prevent this LclVar from being optimized away
    arg->gtOp.gtOp1->gtFlags |= GTF_DONT_CSE;   // the child of a GT_ADDR is required to have this flag set
}